gmso/formats/mcf.py

```diff
@@ -0,0 +1,700 @@
+                improper.atom1.idx + 1,
+                improper.atom2.idx + 1,
+                improper.atom3.idx + 1,
+                improper.atom4.idx + 1,
+                improper_type,
+                improper.type.psi_k * KCAL_TO_KJ,
+                improper.type.psi_eq,
+            )
+        )
+
+
+def _write_fragment_information(mcf, top, frag_list, frag_conn):
+    """Write the fragments in the molecule.
+
+    Parameters
+    ----------
+    mcf : file object
+        The file object of the Cassandra mcf being written
+    top : gmso.core.Topology
+        Topology object
+    frag_list : list
+        Atom ids belonging to each fragment
+    frag_conn : list
+        Fragment ids of connected fragments
+
+    """
+
+    header = (
+        "\n!Fragment Format\n"
+        "!index number_of_atoms_in_fragment branch_point other_atoms\n"
+        "\n# Fragment_Info\n"
+    )
+
+    mcf.write(header)
+
+    # Special cases first
+    if len(frag_list) == 0:
+        if len(top.sites) == 1:
+            mcf.write("1\n")
+            mcf.write("1 1 1\n")
+        elif len(top.sites) == 2:
+            mcf.write("1\n")
+            mcf.write("1 2 1 2\n")
+        else:
+            warnings.warn(
+                "More than two atoms present but " "no fragments identified."
+            )
+            mcf.write("0\n")
+    else:
+        mcf.write("{:d}\n".format(len(frag_list)))
+        for i, frag in enumerate(frag_list):
+            mcf.write("{:d}    {:d}".format(i + 1, len(frag)))
+            for idx in frag:
+                mcf.write("    {:d}".format(idx + 1))
+            mcf.write("\n")
+
+    mcf.write("\n\n# Fragment_Connectivity\n")
+    mcf.write("{:d}\n".format(len(frag_conn)))
+    for i, conn in enumerate(frag_conn):
+        mcf.write(
+            "{:d}    {:d}    {:d}\n".format(i + 1, conn[0] + 1, conn[1] + 1)
+        )
+
+
+def _write_intrascaling_information(mcf, lj14, coul14):
+    """Write the intramolecular scaling in the molecule.
+
+    Parameters
+    ----------
+    mcf : file object
+        The file object of the Cassandra mcf being written
+    lj14 : float
+        The 1-4 scaling parameter for LJ interactions
+    coul14 : float
+        The 1-4 scaling parameter for Coulombic interactions
+
+    """
+
+    header = (
+        "\n!Intra Scaling\n"
+        "!vdw_scaling    1-2 1-3 1-4 1-N\n"
+        "!charge_scaling 1-2 1-3 1-4 1-N\n"
+        "\n# Intra_Scaling\n"
+    )
+
+    mcf.write(header)
+    mcf.write("0. 0. {:.4f} 1.\n".format(lj14))
+    mcf.write("0. 0. {:.4f} 1.\n".format(coul14))
+
+
+def _check_compatibility(top):
+    """Check Topology object for compatibility with Cassandra MCF format"""
+
+    if not isinstance(top, Topology):
+        raise GMSOError("MCF writer requires a Topology object.")
+    if not all([site.atom_type.name for site in top.sites]):
+        raise GMSOError(
+            "MCF writing not supported without " "parameterized forcefield."
```

Comment:
also weird formatting, unless I am missing something

Reply:
Yes, seems to have come from `black`.